addons/sale_stock/models/sale_order.py

```diff
@@ -428,6 +428,13 @@ def write(self, values):
         if 'product_uom_qty' in values:
             lines = self.filtered(lambda r: r.state == 'sale' and not r.is_expense)
 
+        if 'product_packaging_id' in values:
+            moves_to_update = self.env['stock.move']
+            for move_id in self.move_ids:
+                if move_id.product_packaging_id.id != values['product_packaging_id'] and move_id.state not in ['cancel', 'done']:
+                    moves_to_update += move_id
+            moves_to_update.product_packaging_id = values['product_packaging_id']
```

Comment:
```suggestion
```suggestion
            self.move_ids.filtered(
                lambda m: m.state not in ['cancel', 'done']
            ).product_packaging_id = values['product_packaging_id']
```
the check on the difference is not mandatory and the code is much smaller (you can also do the same in purchase_stock
```